apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/GenesisImporter.java

```diff
@@ -14,122 +14,241 @@
+    private final ApplicationJsonFactory jsonFactory;
+    /**
+     * Represents a total number of public keys in a genesisAccounts.json file.
+     * Has a hardcoded value because of the immutability of this file.
+     */
+    private final int publicKeyNumberTotal;
+    /**
+     * Represents a total number of balances in a genesisAccounts.json file.
+     * Has a hardcoded value because of the immutability of this file.
+     */
+    private final int balanceNumberTotal;
+    private final BlockchainConfigUpdater blockchainConfigUpdater;
+    private byte[] CREATOR_PUBLIC_KEY;
+    private final BlockchainConfig blockchainConfig;
+    private final AplAppStatus aplAppStatus;
+    private final DatabaseManager databaseManager;
     private String genesisTaskId;
-    private JsonNode genesisAccountsJSON = null;
-    private List<String> publicKeys;
-    private Map<String, Long> balances;
     private byte[] computedDigest;
-    private ObjectMapper mapper = new ObjectMapper();
-    private String genesisParametersLocation;
+    private final String genesisParametersLocation;
 
     @Inject
-    public GenesisImporter(BlockchainConfig blockchainConfig, ConfigDirProvider configDirProvider,
-                           BlockchainConfigUpdater blockchainConfigUpdater,
-                           DatabaseManager databaseManager,
-                           AplAppStatus aplAppStatus,
-                           GenesisImporterProducer genesisImporterProducer) {
-        this.blockchainConfig = Objects.requireNonNull(blockchainConfig, "blockchainConfig is NULL");
-        this.configDirProvider = Objects.requireNonNull(configDirProvider, "configDirProvider is NULL");
-        this.blockchainConfigUpdater = Objects.requireNonNull(blockchainConfigUpdater, "blockchainConfigUpdater is NULL");
-        this.databaseManager = Objects.requireNonNull(databaseManager, "databaseManager is NULL");
-        this.aplAppStatus = Objects.requireNonNull(aplAppStatus, "aplAppStatus is NULL");
-        Objects.requireNonNull(genesisImporterProducer);
-        this.genesisParametersLocation = genesisImporterProducer.genesisParametersLocation();
+    public GenesisImporter(
+            BlockchainConfig blockchainConfig,
+            BlockchainConfigUpdater blockchainConfigUpdater,
+            DatabaseManager databaseManager,
+            AplAppStatus aplAppStatus,
+            GenesisImporterProducer genesisImporterProducer,
+            ApplicationJsonFactory jsonFactory
+    ) {
+        this.blockchainConfig = getBlockchainConfig(blockchainConfig);
+        this.blockchainConfigUpdater = getBlockchainConfigUpdater(blockchainConfigUpdater);
+        this.databaseManager = getDatabaseManager(databaseManager);
+        this.aplAppStatus = getAplAppStatus(aplAppStatus);
+        this.genesisParametersLocation = getGenesisParametersLocation(genesisImporterProducer);
+        this.jsonFactory = getJsonFactory(jsonFactory);
+        this.publicKeyNumberTotal = 230730;
```

Comment:
These values are changeable and depends on genesis-accounts.json file. Genesis files can be different, but in our case we still did not modify them.

Reply:
These files are always the same for all test-nets now, @alukin approved to hardcode these values

Reply:
Ok, leave it, but reuse constructor, provided below.

Reply:
I moved them to apl-blockchain.properties